src/net.c

```diff
@@ -218,8 +218,8 @@ struct mg_timer *mg_timer_add(struct mg_mgr *mgr, uint64_t milliseconds,
                               unsigned flags, void (*fn)(void *), void *arg) {
   struct mg_timer *t = (struct mg_timer *) calloc(1, sizeof(*t));
   if (t != NULL) {
+    flags |= MG_TIMER_AUTODELETE;  // We have calloc-ed it, so autodelete
```

Comment:
I don't really like defaulting to AUTODELETE.
Flagging a repeating timer as such, has no sense (unless the user manually changes the timer type...)
From this API (Mongoose API, not Timer API) it makes sense to auto-delete, as we do not provide a way to restart a single use timer.
I would only mark as `MG_TIMER_AUTODELETE` when `! MG_TIMER_REPEAT`

Reply:
OTOH, your comment is compelling... whatever the other API does, we take responsibility... both choices are fine with me, let's keep this.